diff --git a/main.cpp b/main.cpp
index c8f39f5..32266c7 100644
--- a/main.cpp
+++ b/main.cpp
@@ -164,15 +164,17 @@ APICALL EXPORT std::string PLUGIN_API_VERSION() { return HYPRLAND_API_VERSION; }
 APICALL EXPORT PLUGIN_DESCRIPTION_INFO PLUGIN_INIT(HANDLE handle) {
   PHANDLE = handle;

-  const std::string HASH = __hyprland_api_get_hash();
-
-  if (HASH != GIT_COMMIT_HASH) {
-    HyprlandAPI::addNotification(
-        PHANDLE,
-        "[Hyprfoci] Failure in initialization: Version mismatch (headers ver "
-        "is not equal to running hyprland ver)",
-        CHyprColor{1.0, 0.2, 0.2, 1.0}, 5000);
-    throw std::runtime_error("[Hyprfoci] Version mismatch");
+  const std::string HASH        = __hyprland_api_get_hash();
+  const std::string CLIENT_HASH = __hyprland_api_get_client_hash();
+
+  if (HASH != CLIENT_HASH) {
+    HyprlandAPI::addNotification(PHANDLE,
+                                 "[hyprsplit] Failure in initialization: Version mismatch (headers "
+                                 "ver is not equal to running hyprland ver)",
+                                 CHyprColor{1.0, 0.2, 0.2, 1.0}, 5000);
+    HyprlandAPI::addNotification(PHANDLE, std::format("[hyprsplit] compositor hash: {}", HASH), CHyprColor{1.0, 0.2, 0.2, 1.0}, 5000);
+    HyprlandAPI::addNotification(PHANDLE, std::format("[hyprsplit] client hash: {}", CLIENT_HASH), CHyprColor{1.0, 0.2, 0.2, 1.0}, 5000);
+    throw std::runtime_error("[hyprsplit] Version mismatch");
   }

   // config variables
